operator/pkg/gateway-api/route_checks.go

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package gateway_api
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// TODO(https://github.com/cilium/cilium/issues/25130): turn this into a generic checker that can be used for all route types
+type backendValidatonFunc func(ctx context.Context, log *logrus.Entry, c client.Client, tr *gatewayv1alpha2.TLSRoute) (ctrl.Result, bool, error)
+
+func checkAgainstCrossNamespaceReferences(ctx context.Context, log *logrus.Entry, c client.Client, tr *gatewayv1alpha2.TLSRoute) (ctrl.Result, bool, error) {
```

Comment:
It _may_ be better to also do ReferenceGrant checking here as well as the other places it gets done; I don't think there are any conformance checks for TLSRoute ReferenceGrants yet, but this may fail.

Maybe a TODO so we can grep for ReferenceGrant later if we need to? Non-blocking for now though.